state-chain/traits/src/lib.rs

```diff
@@ -446,6 +446,53 @@ where
 	}
 }
 
+pub trait CfeEventEmitterForCrypto<T: Chainflip, C: ChainCrypto> {
+	fn keygen_request(
+		ceremony_id: CeremonyId,
+		epoch_index: EpochIndex,
+		participants: BTreeSet<T::ValidatorId>,
+	);
+
+	fn signature_request(
+		ceremony_id: CeremonyId,
+		epoch_index: EpochIndex,
+		key: C::AggKey,
+		signatories: BTreeSet<T::ValidatorId>,
+		payload: C::Payload,
+	);
+
+	fn key_handover_request(
+		_ceremony_id: CeremonyId,
+		_from_epoch: EpochIndex,
+		_to_epoch: EpochIndex,
+		_key_to_share: C::AggKey,
+		_sharing_participants: BTreeSet<T::ValidatorId>,
+		_receiving_participants: BTreeSet<T::ValidatorId>,
+		_new_key: C::AggKey,
+	) {
+		// Do nothing by default
```

Comment:
We could use `key_handover_is_required` which is on `ChainCrypto` here to have a stricter default

Reply:
Not sure if this can be done at compile time (using the current key_handover_is_required(), which is a "runtime" function), but I added an assert to check that key_handover_is_required is false in the default implementation.